TheraLang.BLL/Services/SiteMapService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using TheraLang.BLL.DataTransferObjects;
+using TheraLang.BLL.Interfaces;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.UnitOfWork;
+
+namespace TheraLang.BLL.Services
+{
+    public class SiteMapService:ISiteMapService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public SiteMapService(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+        
+        public async Task<IEnumerable<SiteMapDto>> GetAll()
+        {
+            var entities = await _unitOfWork.Repository<Page>().Get()
+                .Include(sm => sm.SubPages)
+                .OrderBy(sm=>sm.SortOrder)
+                .ToListAsync();
+            
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<Page, SiteMapDto>()
+                .ForMember(dto => dto.SubPages,opts => 
+                    opts.MapFrom(entity => entity.SubPages.OrderBy(subpage=>subpage.SortOrder)))
+            ).CreateMapper();
+            
+            var onlyRoots = mapper.Map<IEnumerable<Page>, IEnumerable<SiteMapDto>>(entities)
+                .Where(sm => sm.ParentPageId == null);
+            
+            return onlyRoots;
+        }
+
+        public async Task UpdateStructure(IEnumerable<SiteMapDto> siteMap)
+        {
+            var siteMapDtos = siteMap.ToList();
+            var idsOfChanged = siteMapDtos.Select(sm => sm.Id).ToList();
+
+            var entitiesToChange = await _unitOfWork.Repository<Page>().Get().Where(p =>

```

Comment:
what about using here GetAll with predicate and you will not need to write where here